src/factories/inputs/FactoryNumberInput.tsx

```diff
@@ -0,0 +1,43 @@
+import { NumberInput, type NumberInputProps } from '@mantine/core';
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+export function FactoryNumberInput(props: NumberInputProps) {
+  // Very complicated yet simple handlers for "shift for +10" behaviour on the resource number input
+  const [inputStep, setInputStep] = useState(1);
+  const keyDownHandler = useCallback(
+    (e: KeyboardEvent) => e.key == 'Shift' && setInputStep(10),
+    [],
+  );
+  const keyUpHandler = useCallback(
+    (e: KeyboardEvent) => e.key == 'Shift' && setInputStep(1),
+    [],
+  );
+  const eventListeners = useRef<{
+    keydown: (e: KeyboardEvent) => void;
+    keyup: (e: KeyboardEvent) => void;
+  }>();
+  useEffect(() => {
+    // Here will be adding the static listener so we can keep the reference
+    eventListeners.current = { keydown: keyDownHandler, keyup: keyUpHandler };
```

Comment:
Hey, what's the point of useRef vs a local variable in the useEffect callback function?